### DISCUSSION TITLE
Smart Paste

### DISCUSSION THREAD

**Discussion by FrancoPaez** [tinymce/tinymce-react#289] [Q&A]
Hi! I am encountering a problem related to pasting of formatted text.

I am testing the editor, and they asked me that it support pasting content from Word, WordOnline, Excel, ExcelOnline and Google Docs, Google Sheets, etc.

The editor responds perfectly, but is there any way that when pasting Excel tables, only those tables will erase the format? and also allow me to send a notification to the user saying that only tables cannot be copied and pasted. I hope you understand my problem

**sample text from Google Docs**
![image](https://user-images.githubusercontent.com/82821596/135678852-f6cbe140-350b-49b0-865e-b55bda059a5c.png)

**result**
![image](https://user-images.githubusercontent.com/82821596/135678901-7840cfa1-b07d-4554-b399-6ed81a3dda39.png)

**expected output**
![image](https://user-images.githubusercontent.com/82821596/135679007-b2a0924d-9340-4176-8a10-edfce9526366.png)

**Comment by EkimChau** [comment]
Hi FrancoPaez

Could you clarify what you mean " is there any way that when pasting Excel tables, only those tables will erase the format?"
I assume you are referring to the 3rd image where the structure is removed?  If you are using our power paste plugin, you can report the issue to https://support.tiny.cloud/ and mention the excel version you were using.

"allow me to send a notification to the user saying that only tables cannot be copied and pasted"
Use TinyMCE's notifications api to send this message (warn, error or info modes) would be the first step.

To implement smart notification, you can use the paste_preprocess function to inspect what is being pasted, if the content contains a table, then fire a notification.  see the docs https://www.tiny.cloud/docs/plugins/opensource/paste/#paste_preprocess

hope that helps :)
